### PR TITLE
[Arm64] Adjust alignment for 128 byte cache line

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
@jkotas @dotnet/arm64-contrib PTAL.

**Comment by jkotas** [comment]
cc @kouvel @jkoritzinsky

**Comment by kouvel** [comment]
@dotnet-bot test Ubuntu arm64 Cross Debug Build

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by kouvel** [comment]
Trying to figure out why the "Ubuntu arm64 Cross Debug Build" job is stuck. If it stays that way for much longer, I'll go ahead and merge as I really don't think this will have any impact in that job.

**Comment by RussKeldorph** [comment]
@jashook Can you give latest status on the Ubuntu ARM64 job?

**Comment by jashook** [comment]
@dotnet-bot Ubuntu arm64 Cross Debug Build

**Comment by jashook** [comment]
It is strange why this has not started. The machine is connected and seems like it should not be blocked. Will look into it.

**Comment by jashook** [comment]
Either way these jobs are still not expected to run green that work has yet to be done. Please do not rely on the arm64 ubuntu runs, use the arm64 windows checked runs for validation for now.

**Comment by kouvel** [comment]
@jashook ok, thanks!

**Comment by kouvel** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build and Test

**Comment by kouvel** [comment]
@dotnet-bot test this please

**Comment by jashook** [comment]
@kouvel arm32 Windows Job should fail with two failures, #12945 fixes this.

**Comment by kouvel** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build and Test

**Comment by kouvel** [comment]
It looks like the Windows arm cross checked build timed out, and is not restarting even with "test this please". Not sure if this is the same issue that was fixed.

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build and Test

**Comment by kouvel** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build and Test

**Comment by kouvel** [comment]
@dotnet-bot test this please

File: src/vm/threadpoolrequest.cpp [hunk 1]
```diff
@@ -30,10 +30,10 @@
 BYTE PerAppDomainTPCountList::s_padding[64 - sizeof(LONG)];
```
Comment (sdmaclea) [inline src/vm/threadpoolrequest.cpp]: @janvorli Fixed

File: src/vm/threadpoolrequest.h [hunk 2]
```diff
@@ -184,10 +184,9 @@ class ManagedPerAppDomainTPCount : public IPerAppDomainTPCount {
     ADID m_id;
     TPIndex m_index;
     DECLSPEC_ALIGN(MAX_CACHE_LINE_SIZE) struct {
-        BYTE m_padding1[64 - sizeof(LONG)];
+        BYTE m_padding1[MAX_CACHE_LINE_SIZE - sizeof(LONG)];
         // Only use with VolatileLoad+VolatileStore+FastInterlockCompareExchange
         LONG m_numRequestsPending;
-        BYTE m_padding2[64];
```
Comment (janvorli) [inline src/vm/threadpoolrequest.h]: I think this second padding is still needed. If I understand its purpose correctly, it was used to ensure that the m_numRequestsPending doesn't share the cacheline with anything even if the ManagedPerAppDomainTPCount was not allocated with alignment (which is actually the case, it is allocated using regular new).

Comment (sdmaclea) [inline src/vm/threadpoolrequest.h]: OK I put all the padding arrays back and added one in ThreadCount to be consistent with UnfairSemaphore

**Comment by sdmaclea** [comment]
@janvorli I have addressed your requested changes, please review at your convenience

chaos65204782cs failed retry. @dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by sdmaclea** [comment]
@janvorli ping

**Pull request merged**